bokeh/io.py

```diff
@@ -413,6 +414,8 @@ def _get_save_args(state, filename, resources, title):
     if filename is None:
         warn = False
         filename = _detect_filename("html")
+        if not os.access(filename, os.W_OK):
```

Comment:
`_detect_filename()` can return `None`, indicating that detection failed, so `os.access()` check should be moved there.